### PR TITLE
AutoJoinChannels WIP

### PR DESCRIPTION
**Pull request by TurBoss**
cleanup of #24 sry

### CODE REVIEW

**Comment by TurBoss** [comment]
channels are only stored when joining a seccond one

**Comment by gajop** [comment]
1) You don't need to invoke self.SaveChannelsInConfig() .. besides you're invoking it wrong (should be self: not self.); Just invoke Configuration:SaveConfig() directly 
2) Fix whitespace -> tabs before merge
3) I'm not sure this will work well if there's a connection issue. Test by using a local server and restarting it.

**Comment by TurBoss** [comment]
seems that all is using tabs

**Comment by gajop** [comment]
still aren't tabs (should be tabs)

File: LuaUI/widgets/chobby/components/chat_windows.lua [hunk 1]
```diff
@@ -301,6 +301,7 @@ function ChatWindows:UpdateChannels(channelsArray)
 					OnClick = {
 						function()
 							lobby:Join(channel.chanName)
+                            Configuration:SaveConfig()
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/chat_windows.lua]: should be tabs here still

File: LuaUI/widgets/chobby/components/configuration.lua [hunk 2]
```diff
@@ -21,29 +22,29 @@ function Configuration:init()
 	self.buttonSelectedColor = {0.54,0.72,1,0.6}--{1.0, 1.0, 1.0, 1.0}
 
 	self.configFile = ".chobby/config.json"
+
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/configuration.lua]: delete whitespace, also don't need "self.configFile" anymore

File: LuaUI/widgets/chobby/components/configuration.lua [hunk 3]
```diff
@@ -21,29 +22,29 @@ function Configuration:init()
...
 			self[k] = v
 		end
 	end
 end
 
 function Configuration:SaveConfig()
-	local out = {}
-	out = {
+
+	self.channels = lobby:GetMyChannels()
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/configuration.lua]: not needed i think, along with `:GetChannels`

File: LuaUI/widgets/chobby/components/configuration.lua [hunk 4]
```diff
@@ -21,29 +22,29 @@ function Configuration:init()
...
+	
+	return {
 		userName = self.userName,
 		password = self.password,
 		autoLogin = self.autoLogin,
+		channels = self.channels,
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/configuration.lua]: just use lobby:GetMyChannels() here

File: LuaUI/widgets/chobby/components/login_window.lua [hunk 5]
```diff
@@ -205,7 +207,7 @@ function LoginWindow:tryLogin()
 	Configuration.password  = password
 	Configuration.autoLogin = self.cbAutoLogin.checked
 	--Configuration.autoLogin = true
-	Configuration:SaveConfig()
+	--Configuration:SaveConfig()
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/login_window.lua]: delete

File: LuaUI/widgets/chobby/components/login_window.lua [hunk 6]
```diff
@@ -297,6 +299,9 @@ function LoginWindow:OnConnected()
 				end
 			end,
 		})
+		for k, v in pairs(self.channels) do
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/login_window.lua]: don't localize self.channels, invoke the function here

File: LuaUI/widgets/chobby/components/status_bar/sb_menu_icon.lua [hunk 7]
```diff
@@ -148,7 +148,7 @@ function SBMenuIcon:Logout()
 		Spring.Echo("Logout")
 		lobby:Disconnect()
 		Configuration.autoLogin = false
-		Configuration:SaveConfig()
+		--Configuration:SaveConfig()
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/status_bar/sb_menu_icon.lua]: delete

File: LuaUI/widgets/chobby/core.lua [hunk 8]
```diff
@@ -120,4 +122,13 @@ function Chobby:_PrintError(err)
 	Spring.Log("Chobby", LOG.ERROR, debug.traceback(err))
 end
 
+function Chobby:_GetConfigData()
+	data = Configuration:SaveConfig()
```
Comment (gajop) [inline LuaUI/widgets/chobby/core.lua]: just type return Configuration:SaveConfig(), don't localize

File: LuaUI/widgets/chobby/core.lua [hunk 9]
```diff
@@ -123,8 +123,7 @@ function Chobby:_PrintError(err)
 end
 
 function Chobby:_GetConfigData()
-	data = Configuration:SaveConfig()
-	return data
+	return Configuration:SaveConfig()
```
Comment (gajop) [inline LuaUI/widgets/chobby/core.lua]: also rename Configuratino's :SaveConfig to :GetConfigData. It does the same thing, just is named differently (might be confusing0

File: LuaUI/widgets/chobby/components/chat_windows.lua [hunk 10]
```diff
@@ -456,3 +456,4 @@ function ChatWindows:CreateJoinChannelWindow()
 	}
 	screen0:FocusControl(ebChannelName)
 end
+
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/chat_windows.lua]: delete this whitespace

File: LuaUI/widgets/chobby/components/configuration.lua [hunk 11]
```diff
@@ -20,30 +21,26 @@ function Configuration:init()
...
 
 function Configuration:SaveConfig()
-	local out = {}
-	out = {
+	
+	return {
 		userName = self.userName,
 		password = self.password,
 		autoLogin = self.autoLogin,
+		channels = lobby:GetMyChannels() ,
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/configuration.lua]: delete the space before teh comma

File: LuaUI/widgets/chobby/components/login_window.lua [hunk 12]
```diff
@@ -195,7 +195,8 @@ end
 
 function LoginWindow:tryLogin()
 	self.lblError:SetCaption("")
-
+
```
Comment (gajop) [inline LuaUI/widgets/chobby/components/login_window.lua]: random whitespace changes, remove them

**Pull request closed without merging**